src/commands/check.rs

```diff
@@ -161,17 +161,24 @@ pub async fn check(args: CheckArgs) -> anyhow::Result<()> {
         };
 
         if !diagnostics.is_empty() {
-            emit_diagnostics(
+            let diagnostics = if suppress {
                 diagnostics
                     .iter()
-                    .filter(|d| !suppress || d.severity() == Severity::Error),
+                    .filter(|d| d.severity() == Severity::Error)
+                    .cloned()
+                    .collect()
+            } else {
+                diagnostics.to_vec()
+            };
```

Comment:
Rather than collecting here, could we simply apply an `if` condition on the match below on `diagnostic.severity()`?

Such as:

```rust
 match diagnostic.severity() {
    Severity::Error => error_count += 1,
    Severity::Warning if !suppress => warning_count += 1,
    Severity::Note if !suppress => note_count += 1,
    _ => {}
}
```